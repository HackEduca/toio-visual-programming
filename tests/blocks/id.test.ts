/**
 * Copyright (c) Sony Interactive Entertainment Inc.
 *
 * This source code is licensed under the BSD-3-Clause license found
 * in the LICENSE file in the root directory of this source tree.
 */

import { initialize, convertFromIntegerToByteArray } from '../util'

let blocks
let notify

beforeEach(async () => {
  const util = await initialize('')
  blocks = util.blocks
  notify = util.notify
})

describe('"getXPosition" block', () => {
  it('gets a x position on a mat correctly', async () => {
    notify(
      1,
      convertFromIntegerToByteArray(200, 2),
      convertFromIntegerToByteArray(300, 2),
      convertFromIntegerToByteArray(45, 2)
    )

    expect(blocks.getXPosition()).toMatchSnapshot()
  })
})

describe('"getYPosition" block', () => {
  it('gets a y position on a mat correctly', async () => {
    notify(
      1,
      convertFromIntegerToByteArray(200, 2),
      convertFromIntegerToByteArray(300, 2),
      convertFromIntegerToByteArray(45, 2)
    )

    expect(blocks.getYPosition()).toMatchSnapshot()
  })
})

describe('"getDirection" block', () => {
  it('gets a direction on a mat correctly', async () => {
    notify(
      1,
      convertFromIntegerToByteArray(200, 2),
      convertFromIntegerToByteArray(300, 2),
      convertFromIntegerToByteArray(45, 2)
    )

    expect(blocks.getDirection()).toMatchSnapshot()
  })
})
