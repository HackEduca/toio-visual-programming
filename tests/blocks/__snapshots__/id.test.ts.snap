// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"getDirection" block gets a direction on a mat correctly 1`] = `135`;

exports[`"getXPosition" block gets a x position on a mat correctly 1`] = `-45`;

exports[`"getYPosition" block gets a y position on a mat correctly 1`] = `-47.220779220779214`;
